### PR TITLE
feat(categories): prefill type from selected tab

### LINKED ISSUE #221: feature: Prefill category type from selected tab
### What problem are you trying to solve?

Users forced to enter category type each time they are creating a new category, even though they 've selected the category type tab before creation.
This is similar to transaction category type prefill implemented in https://github.com/iguliaev/moneylens/pull/212

### Proposed solution

When user selects category tab on Categories page, they've already chosen the category type for a new category they are trying to enter, so the category type selector on a category page should be prefilled with the category taken from the tab. See similar feature implementation in https://github.com/iguliaev/moneylens/pull/212

### Alternatives considered

_No response_

### Primary area

categories

### Expected impact

low

### Who benefits and how?

The feature is expected to speed up a new category entry for all users

### What does success look like?

_No response_

### PR DESCRIPTION
**Pull request by iguliaev**
## Why

Users had to reselect the category type when creating a new category, even after choosing a category tab on the Categories page. This makes the create flow slower than it needs to be. Fixes #221.

## What Changed

- Files or areas updated:
  - `apps/web-next/src/pages/categories/list.tsx`
  - `apps/web-next/src/pages/categories/create.tsx`
  - `apps/web-next/e2e/tests/categories.spec.ts`
  - `apps/web-next/e2e/utils/test-helpers.ts`
- New functionality added:
  - The Categories list Create button now carries the active tab type into `/categories/create`.
  - The category create form preselects Type only when it comes from the Categories list context.
  - The parent category dropdown is driven by the form's current type so it is ready on first render.
  - E2E coverage now checks prefill, blank direct create, invalid params, and dropdown readiness.
- Existing behavior changed:
  - Direct `/categories/create` visits remain blank.
  - The shared category create helper now handles already-prefilled Type values.

## Key Decisions

- Decision: use query params for explicit context handoff.
- Reasoning: it keeps the behavior predictable, debuggable, and aligned with the existing transaction prefill flow.
- Alternatives considered: shared UI state and persistent last-used type, both of which were broader than the issue asked for.

## How This Affects the System

- User-facing changes:
  - Creating a category from a selected tab is faster because Type is prefilled.
- API changes:
  - None.
- Performance implications:
  - None meaningful.
- Database changes:
  - None.
- Breaking changes:
  - None.

## Testing

- New tests added:
  - `create from selected category tab preselects type`
  - `direct create stays blank`
  - `invalid query params stay blank`
- Existing tests status:
  - Full `categories.spec.ts` suite passes.
- Manual testing performed:
  - Verified Create from each tab, direct `/categories/create`, and invalid query params.
- Edge cases tested:
  - Prefilled Type does not break the parent-category dropdown.
- Test files modified or added:
  - `apps/web-next/e2e/tests/categories.spec.ts`
  - `apps/web-next/e2e/utils/test-helpers.ts`

### CODE REVIEW

**Comment by iguliaev** [comment]
Addressed: the helper comment now reflects the `/categories/create` page flow, and the data-reset E2E test now uses `createCategoryForType()` so it follows the same navigation path as the rest of the suite.

File: apps/web-next/e2e/utils/test-helpers.ts [hunk 1]
```diff
@@ -360,16 +360,19 @@ export async function createCategoryForType(
...
+  const escapedType = type.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+  await page.getByRole("combobox", { name: "* Type" }).click({ force: true });
+  const typeOption = page
+    .locator(".ant-select-dropdown:visible")
+    .getByTitle(new RegExp(`^${escapedType}$`, "i"));
+  await typeOption.waitFor({ state: "visible" });
+  await typeOption.click();
   await expect(
     page.locator("#root").getByTitle(new RegExp(type, "i"))
   ).toBeVisible();
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/e2e/utils/test-helpers.ts:378]: Updated to use `escapedType` in the final assertion as well, so the helper now uses the escaped regex value consistently.

File: apps/web-next/src/pages/categories/create.tsx [hunk 2]
```diff
@@ -1,35 +1,68 @@
...
   const { formProps, saveButtonProps } = useForm();
-  const [selectedType, setSelectedType] = useState<string | undefined>();
+  const initialType = useMemo<string | undefined>(() => {
+    const validTypes = new Set<string>(Object.values(TRANSACTION_TYPES));
+    const source = searchParams.get("source");
+    const rawType = searchParams.get("type");
+
+    if (source !== "categories-list") return undefined;
+    if (!rawType || !validTypes.has(rawType)) {
+      return undefined;
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/categories/create.tsx:20]: Keeping this as-is for now; the memo only reruns when search params change, so the allocation is not on the hot path. The type narrowing is a cleanup I can do if we want the extra typing.

File: apps/web-next/e2e/tests/categories.spec.ts [hunk 3]
```diff
@@ -41,6 +41,57 @@ test.describe("Categories", () => {
...
+      .click();
+
+    await page.getByRole("button", { name: "Create" }).click();
+
+    await expect(page).toHaveURL(
+      new RegExp(`/categories/create\\?source=categories-list&type=${type}$`)
+    );
+    await expect(
+      page.locator(".ant-select-selection-item").filter({ hasText: /^earn$/i })
+    ).toBeVisible();
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/e2e/tests/categories.spec.ts:66]: This test is intentionally scoped to the earn case, so I kept the literal assertion explicit. Using the variable would be fine too, but it would not change behavior here.
